src/platform/pdf/templates/medications.js

```diff
@@ -0,0 +1,391 @@
+    introduction.add(createSubHeading(doc, config, data.preface, { x: 16 }));
+  }
+
+  doc.moveDown();
+  introduction.end();
+};
+
+const generateResultsMedicationListContent = async (
+  medication,
+  doc,
+  results,
+  hasHorizontalRule,
+) => {
+  // medication header
+  if (medication.header) {
+    results.add(
+      await createHeading(doc, 'H3', config, medication.header, {
+        paragraphGap: 10,
+        x: 16,
+      }),
+    );
+  }
+
+  // medication section header
+  for (const section of medication.sections) {
+    results.add(
+      await createHeading(doc, 'H4', config, section.header, {
+        paragraphGap: 10,
+        x: 16,
+      }),
+    );
+
+    // medication section items
+    for (const resultItem of section.items) {
+      let structs;
+      // image item
+      // TODO: check integration when CORS issue is resolved
+      if (resultItem.value && typeof resultItem.value === 'object') {
+        structs = await createImageDetailItem(doc, config, 32, resultItem);
+        // regular item
+      } else {
+        structs = await createDetailItem(doc, config, 32, resultItem);
+      }
+
+      // TODO: refactor to make it work with images when available
+      // If the next item does not fit - move to the next page
+      let height = doc.heightOfString(
+        `${resultItem.title}: ${resultItem.value}`,
+        {
+          font: config.text.font,
+          size: config.text.size,
+        },
+      );
+      height = resultItem.inline ? height : height + 24;
+      if (doc.y + height > doc.page.height - doc.page.margins.bottom)
+        await doc.addPage();
+
+      for (const struct of structs) {
+        results.add(struct);
+      }
+    }
+
+    doc.moveDown(0.5);
+  }
+
+  // horizontal line
+  if (hasHorizontalRule) {
+    results.add(
+      doc.struct('Artifact', () => {
+        addHorizontalRule(doc, 16, 0, 1);
+      }),
+    );
+  }
+};
+
+const generateResultsContent = async (doc, parent, data) => {
+  for (const resultItem of data.results) {
+    const results = doc.struct('Sect', {
+      title: resultItem.header || 'Results',
+    });
+    parent.add(results);
+
+    // results --> header
+    if (resultItem.header) {
+      results.add(
+        createHeading(doc, 'H2', config, resultItem.header, {
+          paragraphGap: 12,
+          x: 16,
+        }),
+      );
+    }
+
+    // results --> preface
+    if (resultItem.preface) {
+      results.add(
+        createSubHeading(doc, config, resultItem.preface, {
+          paragraphGap: 12,
+          x: 16,
+        }),
+      );
+    }
+
+    // results --> items
+    const hasHorizontalRule = resultItem.sectionSeparators !== false;
+    for (const medication of resultItem.medicationsList) {
+      await generateResultsMedicationListContent(
+        medication,
+        doc,
+        results,
+        hasHorizontalRule,
+      );
+    }
+
+    results.end();
+  }
+};
+
+const generateHeaderBanner = async (doc, header, data) => {
```

Comment:
Could you move this and any other copy-pasted functions to utils and update the MR template as well?

Reply:
Will remove TODOs in a separate PR. Thank you.